src/dao/RegistrationDAO.js

```diff
@@ -527,6 +548,89 @@ async function getUseTermsOfAgree (userId, challengeId, roleId) {
   return helper.queryDataFromDB(QUERY_GET_CHALLENGE_TERMS_OF_USE, [userId, challengeId, roleId])
 }
 
+/**
+ * Register component inquiry
+ * @param {Number} userId the user id
+ * @param {Number} challengeId the challenge id
+ * @returns The component info object
+ */
+async function registerComponentInquiry (userId, challengeId) {
+  const [compInfo] = await getComponentInfo(challengeId)
+  if (!compInfo) {
+    throw new Error(`Challenge with id: ${challengeId} does not exist`)
+  }
+  const [userRating] = await getUserRating(userId, compInfo.projectCategoryId + 111)
+  const rating = null
+  if (userRating) {
+    rating = userRating.rating
+  }
+  const nextId = await getCompInquirySeqNextId()
+  await insertRegistrationRecord(
+    nextId,
+    compInfo.componentId,
+    userId,
+    compInfo.comments,
+    1,
+    rating,
+    parseInt(compInfo.projectCategoryId, 10) === DESIGN_PROJECT_TYPE || parseInt(compInfo.projectCategoryId, 10) == DEVELOPMENT_PROJECT_TYPE ? compInfo.projectCategoryId : null,
+    userId,
+    compInfo.version,
+    challengeId
+  )
+  await updateComponentVersionInfo(compInfo.componentVersionId, rating)
+  const [updated] = await getComponentInfo(challengeId)
```

Comment:
this is not needed

Reply:
It is needed because we need the updated record formatted the same way as in the `getComponentInfo`